Add user survey count and average, and a WAU chart to the site-admin …

dadlerj opened the pull request:
…Overview page

Fixes https://github.com/sourcegraph/sourcegraph/issues/1014

Adds survey results and usage stats to the site-admin overview page. @sqs I know eventually we'll be contributing this usage chart via extension, but this is a quick and easy addition using the existing chart logic. Okay to add this now?

How it looks (with 5+ surveys submitted recently): 
![image](https://user-images.githubusercontent.com/5589410/48583111-2ae60680-e8db-11e8-80fd-219b925c90b0.png)


If <5 have been submitted recently:
![image](https://user-images.githubusercontent.com/5589410/48583199-59fc7800-e8db-11e8-8ab8-d9d5e3f2ed61.png)


> This PR updates the CHANGELOG.md file to describe any user-facing changes.

francisschmaltz commented:
@dadlerj Would this be hidden by default?

dadlerj commented:
@francisschmaltz no, why would it be?

francisschmaltz commented:
It should be hidden and an admin could select to make it expand, but by default it should just be a line like the other.

dadlerj commented:
But again, why? The whole point is to actually surface/communicate this, so making it visible is exactly the point of the PR!

francisschmaltz commented:
Moving it into the main page is super useful for at-a-glance info, but there isn't a need to always have bar graphs. Just a single line that says "531 Weekly active users" is enough for 90% of the interactions and clicking a button to expand it would be enough for the rest. I plan to add in an on boarding flow here and it will follow the same. I think this page will see more use in the future and we should strive to make it consistent.

dadlerj commented:
Cool, do you have any prelim layouts/designs for the page? A wireframe would be super helpful to get a sense for what it'll look like for the onboarding project.

Two comments:

1) at a high-level, let's not let pursuit of consistency stand in the way of effective communication. _If_ it's actually useful for our users to have something at their fingertips (and in this case, I can definitely be convinced that it's not necessary — e.g. how I put the survey responses in a line), then our layouts should make it easier, not block it!

2) in this specific situation, count of weekly users on a line doesn't really work — Sourcegraph's backend doesn't provide "users in the last 7 days", it instead calculates "users per calendar week". Which means the count will grow from Monday-->Sunday. This means that we would be, unfortunately, communicating incorrect numbers if we just put a snapshot in... I guess we could use "users last week", or build out new functionality to provide "last 7 days" data (though then we'd probably want to add it to the usage stats page too, to be consistent), but this wouldn't exactly work. There's also a sense in my head that the trend is more interesting than the absolutes, but I'm not totally sure about that piece.

francisschmaltz commented:
@dadlerj Here is a quick wireframe. Clicking the button would expand/collapse the section 
![image](https://user-images.githubusercontent.com/11583013/48584384-70f09980-e8de-11e8-8017-cf68fbba3f2c.png). We could also have another link to somewhere else with more info (like the explore page).

> At a high-level, let's not let pursuit of consistency stand in the way of effective communication.

Effective design and communication should be consistent by design 😸 

> In this specific situation, count of weekly users on a line doesn't really work — Sourcegraph's backend doesn't provide "users in the last 7 days"

For most admins, having a quick stat to give procurement/management would be super useful – especially to give quick reports.

dadlerj commented:
Cool, I can see what you're going for. When I was putting this together I was definitely aware of the fact that this would stand out a bit more, but I had come to believe that the tradeoff of having this data visible without effort is actually quite useful (we hear from people often who don't realize it exists). 

Given that, would your design allow for one of the headers to be auto-expanded by default, but to still be retractable?

dadlerj commented:
Chatted live — next steps for this PR:

1) Add expand functionality to the lines (selectively)
2) For now, have the usage stats line expanded by default
3) When the onboarding project https://github.com/sourcegraph/sourcegraph/issues/975 is complete, the usage stats line will only be expanded by default once all onboarding steps are complete (before that, only the onboarding steps line will be expanded by default)

sqs commented:
Comment on minor point:

> @sqs I know eventually we'll be contributing this usage chart via extension

This doesn't need to come from an extension ever. Maybe it will be contributed using the same internal APIs that extensions eventually can use to contribute stuff, but it seems fine for it to be in the core codebase and run in the same thread (ie not a Web Worker).

dadlerj commented:
@francisschmaltz here's a video of how this looks now. Let me know what you think generally. I know there are a few visual bugs (e.g. the legend from the chart overlapping the button at certain zooms). https://cl.ly/4497840d1af0

A few questions:
* should there be any consistent place for links out to the pages once a section is expanded? Right now I just have that "(see more)" link in the header of the chart, but it's not easy to find...
* what other UI updates do we need on the lines above it?

dadlerj commented:
Great, agreed. I'll get the UI issues fixed and this merged tomorrow.

dadlerj commented:
@francisschmaltz I also added that icon (https://materialdesignicons.com/icon/open-in-new) to each of the buttons that opens up a new page.

And updated the layout like this:

While closed:
![image](https://user-images.githubusercontent.com/5589410/48790936-5420e100-eca5-11e8-8829-3a5f137fe31a.png)

While open:
![image](https://user-images.githubusercontent.com/5589410/48790944-5aaf5880-eca5-11e8-931b-bab0cf3ca2da.png)

Going to merge when it passes CI, but let me know if you want any tweaks after that.
